### ISSUE TITLE
Interrupt functions don't properly restore accumulator and index registers

### ISSUE THREAD

**Issue by Garydos** [KarolS/millfork#19]
As the title states, there's a problem with interrupt functions in that they don't restore the registers
even though they do store them at the beginning of the function.  

Take the compiled `on_nmi` function found in `nes_routines.mfk` for example, the registers are stored onto the stack, but never restored, simply pulled off the stack and thrown away:

```
on_nmi:
	PHA
	TXA
	PHA
	TYA
	PHA
	CLD
	LDA __reg
	PHA
	LDA $0017
	PHA
	LDA $0018
	PHA
	LDA $0019
	PHA
	JSR nmi
	PLA
	STA $0019
	PLA
	STA $0018
	PLA
	STA $0017
	PLA
	STA __reg
	PLA
	PLA
	PLA
	RTI
```

**Comment by KarolS** [comment]
Thanks for the vca

**Comment by KarolS** [comment]
Thanks for the catch. I don't use interrupts much myself, so I missed it. The fix is done, it will be included in the 0.3.14 release.
